### ISSUE TITLE
Syntactic sugar not able to fix syntactic sugar

### ISSUE THREAD

**Issue by mhhashim** [realm/SwiftLint#2876]
the folllowing line gives syntactic sugar warning
var temp = Dictionary<Int, Array<String>>()

**Comment by marcelofabri** [comment]
@mhhashim This is not valid Swift code: you need to specify the type of `Array`:

```
swift
Welcome to Apple Swift version 5.0.1 (swiftlang-1001.0.82.4 clang-1001.0.46.4).
Type :help for assistance.
  1> var temp = Dictionary<Int, Array>()
error: repl.swift:1:12: error: generic parameter 'Element' could not be inferred
var temp = Dictionary<Int, Array>()
           ^

repl.swift:1:28: note: explicitly specify the generic arguments to fix this issue
var temp = Dictionary<Int, Array>()
                           ^
                                <Any>
```

**Comment by HashimPhilips** [comment]
sorry my bad the actual code was the following which produces the warning
     var sectionsInfo1 = Dictionary<Int, Array<Int>>()

**Comment by marcelofabri** [comment]
This rule is enforcing that you should use the "short" version of the types:

```swift
let foo = [Int: [Int]]()
```
